amr-wind/equation_systems/icns/source_terms/DragForcing.cpp

```diff
@@ -0,0 +1,120 @@
+#include "amr-wind/equation_systems/icns/source_terms/DragForcing.H"
+#include "amr-wind/utilities/IOManager.H"
+
+#include "AMReX_Gpu.H"
+#include "AMReX_Random.H"
+#include "amr-wind/wind_energy/ABL.H"
+
+namespace amr_wind::pde::icns {
+
+DragForcing::DragForcing(const CFDSim& sim)
+    : m_sim(sim)
+    , m_mesh(sim.mesh())
+    , m_velocity(sim.repo().get_field("velocity"))
+    , m_terrainBlank(sim.repo().get_field("terrainBlank"))
+    , m_terrainDrag(sim.repo().get_field("terrainDrag"))
+
+{
+    const auto& abl = m_sim.physics_manager().get<amr_wind::ABL>();
+    const VelPlaneAveraging& fa_velocity =
+        abl.abl_statistics().vel_profile_coarse();
+    gpu_vel_ht.resize(fa_velocity.line_centroids().size());
+    gpu_vel_vals.resize(fa_velocity.line_average().size());
+    amrex::Gpu::copy(
+        amrex::Gpu::hostToDevice, fa_velocity.line_centroids().begin(),
+        fa_velocity.line_centroids().end(), gpu_vel_ht.begin());
+    amrex::Gpu::copy(
+        amrex::Gpu::hostToDevice, fa_velocity.line_average().begin(),
+        fa_velocity.line_average().end(), gpu_vel_vals.begin());
+    /*for(int ii=0;ii<gpu_vel_ht.size();++ii)
+      amrex::Print()<<"H:"<<gpu_vel_ht[ii]
+                    <<" u:"<<gpu_vel_vals[3*ii]
+                    <<" v:"<<gpu_vel_vals[3*ii+1]
+                    <<" w:"<<gpu_vel_vals[3*ii+2]
+                    <<std::endl;*/
+}
+
+DragForcing::~DragForcing() = default;
+
+void DragForcing::operator()(
+    const int lev,
+    const amrex::MFIter& mfi,
+    const amrex::Box& bx,
+    const FieldState fstate,
+    const amrex::Array4<amrex::Real>& src_term) const
+{
+    const auto& vel =
+        m_velocity.state(field_impl::dof_state(fstate))(lev).const_array(mfi);
+    const auto blank = m_terrainBlank(lev).const_array(mfi);
+    const auto drag = m_terrainDrag(lev).const_array(mfi);
+    const auto& geom_vec = m_mesh.Geom();
+    const auto& geom = geom_vec[lev];
+    const auto& dx = geom.CellSize();
+    const auto& prob_lo = geom.ProbLoArray();
+    const auto& prob_hi = geom.ProbHiArray();
+    // const amrex::Real dy = geom.CellSize()[1];
+    // const amrex::Real dz = geom.CellSize()[2];
+    // Real xi = (x - xhi_sponge_start) / (ProbHiArr[0] - xhi_sponge_start);
+    // rho_u_rhs(i, j, k) -= sponge_strength * xi * xi * (rho_u(i, j, k) -
+    // sponge_density*sponge_x_velocity);
+    const amrex::Real gpu_drag = m_drag;
```

Comment:
when copying `m_` stuff locally for use on device, we typically just remove the `m_`. Can you do that?